### PR TITLE
TASK: Further adjustments to Doctrine Migrations integration

### PR DESCRIPTION
**Pull request by kdambekalns**
See https://github.com/neos/flow-development-collection/issues/2244

### CODE REVIEW

File: Neos.Flow/Classes/Persistence/Doctrine/Service.php [hunk 1]
```diff
@@ -84,6 +86,11 @@ class Service
      */
     protected $environment;
 
+    /**
+     * @var BufferedOutput
+     */
+    protected BufferedOutput $logMessages;
```
Comment (bwaidelich) [inline Neos.Flow/Classes/Persistence/Doctrine/Service.php:92]: This ain't PHP 7.3 compatible, is it?

**Review by bwaidelich — CHANGES_REQUESTED** [review]
Great improvement!
But PHP 7.3 doesn't yet support type hints for properties I think

File: Neos.Flow/Classes/Persistence/Doctrine/Service.php (hunk 1 above)
Comment (bwaidelich) [inline Neos.Flow/Classes/Persistence/Doctrine/Service.php]:
```suggestion
    protected $logMessages;
```

**Review by bwaidelich — APPROVED** [review]
Great (by reading)
Will test asap if it doesn't get merged before :)

**Comment by kdambekalns** [comment]
Uhm… the Behat tests fail with

`PHP Fatal error:  Declaration of Symfony\Component\Console\Output\BufferedOutput::doWrite(string $message, bool $newline) must be compatible with Symfony\Component\Console\Output\Output::doWrite($message, $newline) in /home/travis/build/neos/flow-development-distribution/Packages/Libraries/symfony/console/Output/BufferedOutput.php on line 45`

🤔

**Comment by kdambekalns** [comment]
Probably caused by Behat using Console v4.2.2 vs the 5.2.0 Flow installs.

**Comment by kdambekalns** [comment]
Opened https://github.com/neos/behat/issues/21

**Comment by kdambekalns** [comment]
https://github.com/neos/behat/pull/22

**Comment by bwaidelich** [comment]
@kdambekalns thanks! with neos/behat#21 resolved should I trigger a rebuild?

**Comment by kdambekalns** [comment]
First Behat passes, second now running. Static analysis failing unrelated on a changed constant…

**Review by albe — COMMENTED** [review]
Looks good now (tests green)

**Review by bwaidelich — APPROVED** [review]
I just tweaked this a tiny bit by extracting the version normalization into its dedicated method to reduce duplication.
Still experiencing bugs but not related to this PR

**Pull request merged**